src/main/java/net/masterthought/cucumber/util/StepNameFormatter.java

```diff
@@ -31,17 +31,29 @@ public static String format(String stepName, Argument[] arguments, String preArg
 
     private static void surroundArguments(Argument[] arguments, String preArgument, String postArgument, String[] chars) {
         for (Argument argument : arguments) {
-            if (argument.getOffset() == null) {
+            if (isNotMatchedArgument(argument)) {
                 continue;
             }
 
             int start = argument.getOffset();
             int end = start + argument.getVal().length() - 1;
+            if (isArgumentAtEndOfString(start, chars)) {
+                continue;
+            }
+
             chars[start] = preArgument + chars[start];
             chars[end] = chars[end] + postArgument;
         }
     }
 
+    private static boolean isNotMatchedArgument(Argument argument) {
+        return argument.getOffset() == null;
+    }
+
+    private static boolean isArgumentAtEndOfString(int start, String[] chars) {
+        return start >= chars.length;
```

Comment:
what if the val is not empty but should be added at the end of the step?

Reply:
That's testcase format_endsWithArguments.

If the argument is not empty, then the start must be before the end of string.